src/py42/clients/util.py

```diff
@@ -2,14 +2,14 @@
 
 
 def get_all_pages(func, key, *args, **kwargs):
-
-    item_count = settings.items_per_page
-    page_num = 0
-    while item_count >= settings.items_per_page:
-        page_num += 1
-        response = func(
-            *args, page_num=page_num, page_size=settings.items_per_page, **kwargs
-        )
+    if kwargs.get("page_num") is None:
+        kwargs["page_num"] = 0
+    if kwargs.get("page_size") is None:
+        kwargs["page_size"] = settings.items_per_page
+    item_count = kwargs["page_size"]
+    while item_count >= kwargs["page_size"]:
+        kwargs["page_num"] += 1
```

Comment:
is this right? if `get_all_pages` receives a `page_num` of 1, it will increase it to 2 before it makes its first request.

Reply:
I'm not sure I understand why this function was changed at all, can you explain?

Reply:
Before, in detection lists' paging, the actual parameter on the `func` arg for `page_size` did nothing. This is because it is hardcoded to `100` inside the function. If you _did_, however, try to use the `page_size` parameter on `func` from detectionlists, it still wouldn't matter. This is because this util method here, `get_all_pages()`, uses a `page_size` of `settings.items_per_page` and I would see cases where it would use that over anything from `kwargs`.  The same goes for `page_num`. It would _always_ start at zero unless hardcoded in the actual `func`, like it is in alerts.  BUT, to create a `get_page()` method, we need it to actually be a parameter. Else, it always gets the first page, or in the case of alerts, gets the second page.

Reply:
You are right about your first comment though, weird because I was testing this before and didn't notice it but now I do

Reply:
Trying to make the `page_size` passed twice error not occur... it is passed in from `detectionlists.get_all()` in the kwargs as well as in `get_all_pages()` hardcoded

Reply:
@unparalleled-js something seems still not right here -- we are alway increasing `page_num` by one before the first request gets made, won't this result in the first page always being skipped? Shouldn't we change this so that `response = func(*args, **kwargs)` happens first and _then_ the page num gets increased?

Reply:
Maybe you are not seeing it, but it has changed back to it's original impl as far as page num goes. Page size is that only change to this function now, and I have tested all the methods

Reply:
Note that technically, we have _always_ been increasing page num by one before the first request. Look at the main branch.